pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -0,0 +1,278 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package nsxserviceaccount
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	mpmodel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/util"
+)
+
+const (
+	siteId             = "default"
+	enforcementpointId = "default"
+	PortRestAPI        = "rest-api"
+	PortNSXRPCFwdProxy = "nsx-rpc-fwd-proxy"
+	SecretSuffix       = "-nsx-cert"
+	SecretCAName       = "ca.crt"
+	SecretCertName     = "tls.crt"
+	SecretKeyName      = "tls.key"
+)
+
+var (
+	log       = logger.Log
+	Converter *bindings.TypeConverter
+
+	isProtectedTrue = true
+	vpcRole         = "ccp_internal_operator"
+	readerPath      = "/"
+	readerRole      = "cluster_info_reader"
+
+	antreaClusterResourceType = "AntreaClusterControlPlane"
+	revision1                 = int64(1)
+)
+
+func init() {
+	Converter = bindings.NewTypeConverter()
+	Converter.SetMode(bindings.REST)
+}
+
+type NSXServiceAccountService struct {
```

Comment:
Recently, we have a refactor for store resource, we defined a common resource store. 
 // ResourceStore is the store for resource, embed it to subclass
type ResourceStore struct {
	cache.Indexer        // the ultimate place to store the resource
	bindings.BindingType // used by converter to convert the resource
}

and user can to implemented specific store in own service like
type SecurityPolicyService struct {
	common.Service
	common.Service
	securityPolicyStore *SecurityPolicyStore
	ruleStore           *RuleStore
	groupStore          *GroupStore
}
}

So, we don't call cache.Indexer directly as a store, instead, we can encapsulate calling of cache.Indexer into
store's GetByKey and GetByIndex. 
You can refer to this PR:
https://github.com/vmware-tanzu/nsx-operator/pull/148
do you think you need to do this refactor as well?

Reply:
https://github.com/vmware-tanzu/nsx-operator/pull/152, I have rebase recent vpc changes to main branch. So, right now, main branch include all changes. Give timeline constraints, I will be fine to keep current implementation.